aep/general/0137/aep.md.j2

```diff
@@ -0,0 +1,143 @@
+# Apply
+
+In REST APIs, it is customary to make a `PUT` request to a collection's URI (for
+example, `/v1/publishers/{publisher}/books`) in order to create or update a new
+resource within that collection.
+
+Resource-oriented design (AEP-121) honors this pattern through the `Apply`
+method. These RPCs accept the parent collection and the resource to create or
+update, and return the final resource.
+
+## Guidance
+
+APIs **should** provide an apply method for resources unless it is not valuable
+for users to do so.
+
+### Operation
+
+Apply methods are specified using the following pattern:
+
+- The HTTP verb **must** be `PUT`.
+- Some resources take longer to be applied than is reasonable for a regular API
+  request. In this situation, the API **should** use a [long-running
+  operation](/long-running-operations).
+
+{% tab proto %}
+
+{% sample '../example.proto', 'rpc ApplyBook' %}
+
+- The RPC's name **must** begin with the word `Apply`. The remainder of the
+  RPC name **should** be the singular form of the resource being applied.
+  - The request message **must** match the RPC name, with a `Request` suffix.
+- The collection's parent resource **must** be called `parent`, and
+  **should** be the only variable in the URI path.
+  - The collection identifier (`books` in the above example) **must** be a
+    literal string.
+- There **must** be a `body` key in the `google.api.http` annotation, and it
+  **must** map to the resource field in the request message.
+  - All remaining fields **should** map to URI query parameters.
+- There **should** be exactly one `google.api.method_signature` annotation,
+  with a value of `"parent,{resource},id"`, or "`"parent,{resource}"` if the
+  resource ID is not required.
+- The operation **must** have [strong consistency][].
```

Comment:
SGTM - I'll do a follow-up to move this guidance above the tabs (because it's hard to differentiate common guidance and the more specific guidance otherwise).

Reply:
Is there a reason it needs to be done in a separate PR?